config/config.go

```diff
@@ -256,6 +253,43 @@ func (c *Auth) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+type RegexpExtract struct {
+	Regex Regexp `yaml:"regex"`
+	Value string `yaml:"value"`
+
+	XXX map[string]interface{} `yaml:",inline"`
+}
+
+func (c *RegexpExtract) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type plain RegexpExtract
+	if err := unmarshal((*plain)(c)); err != nil {
+		return err
+	}
+	if err := CheckOverflow(c.XXX, "module"); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Regexp encapsulates a regexp.Regexp and makes it YAML marshalable.
+type Regexp struct {
+	*regexp.Regexp
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (re *Regexp) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Comment:
Make sure to add an unmarshalling test for a config that's missing the regex, we've had issues there in the past